### DISCUSSION TITLE
metabase configed not work

### DISCUSSION THREAD

**Discussion by flyphant** [apache/superset#21588] [Q&A / Help] [answered]
i edit the metabase configuration in the superset/config.py，SQLALCHEMY_DATABASE_URI change to my mysql database。but when use command
`docker-compose -f docker-compose-non-dev.yml up`
the config is:
`
"#" The SQLAlchemy connection string.
"#" SQLALCHEMY_DATABASE_URI = "sqlite:///" + os.path.join(DATA_DIR, "superset.db")
SQLALCHEMY_DATABASE_URI = 'mysql+pymysql://username:password@ip/databasename'

`
**when i start my superset by docker-compose, but my new database didn't work**
superset version: 2.0.0
<img width="966" alt="image" src="https://user-images.githubusercontent.com/11854226/192274848-25057b6f-c12c-46b9-ab5c-8384cbcab260.png">

**Comment by sfirke** [comment, marked as answer]
If you edited `config.py`, it could be overridden by values in `superset_config.py` or `superset_config_docker.py`.  Try it in that file?  I do my config edits in `superset_config_docker.py` as it overrides the others when there's a conflict.

**Reply by flyphant** [reply]
thank you very much! I checked the source code of my superset and found that the config it's truely overwrite by the superset_config.py。I also have a question，will the meta database of superset droped or deleted by db initialization or docker initialization and other scences?

**Reply by sfirke** [reply]
Glad it worked!

No, I think you should be able to run commands like `superset init` and `docker-compose up` and `down` and everything without messing up your metadata db.  I have not had any issues there.  And it sounds like you are using a database on a server, rather than a docker container & volume.  That said, I would make sure you have a backup running on your MySQL server and also every time you are going to mess significantly with your Superset instance (e.g., upgrade to a newer version), first make a copy of your MySQL db.

**Reply by flyphant** [reply]
Thanks again! 
You said as I thought, but I didn't find similar documentation on the official website. Now that you said that, I feel at ease now, thank you ！
